### PR TITLE
GH-1670: As a developer I want VSCode LSP extension to support multiple VSCode instances

### LINKED ISSUE #1670: As a developer I want VSCode LSP extension to support multiple VSCode instances
use std-io instead of tcp

### PR DESCRIPTION
**Pull request by mmews-n4**
closes #1670 

also adds syntax highlighting for package.json files

### CODE REVIEW

File: lsp-clients/n4js-vscode-extension/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"name": "n4js-vscode-extension",
 	"displayName": "N4JS VSCode extension",
 	"description": "Language support for N4JS",
-	"version": "0.19.4",
+	"version": "0.19.3",
```
Comment (szarnekow) [inline lsp-clients/n4js-vscode-extension/package.json]: Downgrade?

File: lsp-clients/n4js-vscode-extension/package.json [hunk 2]
```diff
@@ -63,6 +63,24 @@
 					"command": "n4js.rebuild"
 				}
 			]
+		},
+		"configuration": {
```
Comment (szarnekow) [inline lsp-clients/n4js-vscode-extension/package.json]: Nice!

File: lsp-clients/n4js-vscode-extension/package.json [hunk 3]
```diff
@@ -63,6 +63,24 @@
...
+				"n4js.traceOutput": {
+					"type": "string",
+					"default": "off",
+					"enum": [
+						"off",
+						"verbose"
+					],
+					"enumDescriptions": [
+						"No tracing data is displayed.",
+						"Tracing data is displayed on the Output of N4JS language server"
```
Comment (szarnekow) [inline lsp-clients/n4js-vscode-extension/package.json]: Output -> output

File: lsp-clients/n4js-vscode-extension/src-gen/extensionProvider.js [hunk 4]
```diff
@@ -121,4 +152,11 @@ async function sleep(ms) {
 		setTimeout(resolve, ms);
 	});
 }
+function serialize(object) {
+	let output = '';
+	for(const property in object) {
```
Comment (szarnekow) [inline lsp-clients/n4js-vscode-extension/src-gen/extensionProvider.js]: Is this for debugging? Can we use `JSON.stringify(object, null, 2)` here?

File: lsp-clients/n4js-vscode-extension/src/extensionProvider.n4js [hunk 5]
```diff
@@ -9,35 +9,64 @@
...
+
+
+            let writer: Object;
+            let reader: Object;
+            const socket = await connectToRunningN4jsLspServer(PORT, outputChannel);
+            if (socket) {
+                writer = socket;
+                reader = socket;
+            } else {
+                outputChannel.appendLine('Start new n4js lsp server.');
```
Comment (szarnekow) [inline lsp-clients/n4js-vscode-extension/src/extensionProvider.n4js]: n4js lsp -> N4JS LSP?

File: lsp-clients/n4js-vscode-extension/package.json (hunk 3 above)
Comment (mmews-n4) [reply lsp-clients/n4js-vscode-extension/package.json]: Thanks, I changed the wording since I intended to refer to the Output View of VSCode.

**Comment by mmews-n4** [comment]
Thanks!

**Pull request merged**